src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -55,7 +55,10 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         commandResult = command.execute(model);
 
         saveBooks();
-        undoRedoStack.push(command);
+
+        if (!commandResult.isShowAdd() && !commandResult.isShowEdit() && !commandResult.isShowHelp()) {
+            undoRedoStack.push(command, commandText);
```

Comment:
Good and clean way of fixing bug. But I think showHelp don't need to check since it is not `RedoableCommand` anyways.

Reply:
Yep, echo what @lawwm said. No need to check for `isShowHelp()`.